File: data/StarCoder2/readable_format_HPEC/fortran/file227.f90
subroutine kernel_gemm(ni, nj, nk, alpha, beta, c, a, b)
implicit none

DATA_TYPE, dimension(nk, ni) :: a
DATA_TYPE, dimension(nj, nk) :: b
DATA_TYPE, dimension(nj, ni) :: c
DATA_TYPE :: alpha, beta
integer :: ni, nj, nk
integer :: i, j, k

!$pragma scop
do i = 1, _PB_NI
do j = 1, _PB_NJ
c(j, i) = c(j, i) * beta
do k  = 1, _PB_NK
c(j, i) = c(j, i) + (alpha * a(k, i) * b(j, k))
end do
end do
end do
!$pragma endscop
end subroutine

File: data/llama3_8B_ins_results/readable_format_HPEC/cpp/file227.cpp
#include <iostream>
#include <vector>

using namespace std;

void kernel_gemm(int ni, int nj, int nk, double alpha, double beta, vector<vector<double>>& c, const vector<vector<double>>& a, const vector<vector<double>>& b) {
  for (int i = 0; i < ni; i++) {
    for (int j = 0; j < nj; j++) {
      c[j][i] *= beta;
    }
  }
  for (int i = 0; i < ni; i++) {
    for (int j = 0; j < nj; j++) {
      for (int k = 0; k < nk; k++) {
        c[j][i] += alpha * a[k][i] * b[j][k];
      }
    }
  }
}
